src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisher.java

```diff
@@ -226,12 +226,54 @@ ClientFactory getClientFactory(Jenkins context, String credentialsId) throws Abo
       return this.clientFactory;
     }
 
+    public ListBoxModel doFillClusterNameItems(
+        @AncestorInPath Jenkins context,
+        @QueryParameter("credentialsId") final String credentialsId,
+        @QueryParameter("projectId") final String projectId,
+        @QueryParameter("zone") final String zone) {
+      ListBoxModel items = new ListBoxModel();
+      items.add(EMPTY_NAME, EMPTY_VALUE);
+      if (Strings.isNullOrEmpty(credentialsId)
+          || Strings.isNullOrEmpty(projectId)
+          || Strings.isNullOrEmpty(zone)) {
+        return items;
+      }
+
+      ClientFactory clientFactory;
+      try {
+        clientFactory = getClientFactory(context, credentialsId);
+      } catch (AbortException ae) {
+        LOGGER.log(Level.SEVERE, Messages.KubernetesEnginePublisher_CredentialAuthFailed(), ae);
+        items.clear();
+        items.add(Messages.KubernetesEnginePublisher_CredentialAuthFailed(), EMPTY_VALUE);
+        return items;
+      }
+
+      try {
+        ContainerClient client = clientFactory.containerClient();
+        List<Cluster> clusters = client.listClusters(projectId, zone);
+
+        if (clusters.isEmpty()) {
+          return items;
+        }
+
+        clusters.forEach(c -> items.add(c.getName()));
+        items.get(1).selected = true;
```

Comment:
This approach is causing an existing value to get changed to the first item in the list each time the config page is reloaded. Suggest adding zone as a query param input for the method, and defaulting to selecting the existing value if it exists. Same suggestion for all of the other doFill methods.

Reply:
Done, let me know what you think of the helper method.